bioptim/gui/graph.py

```diff
@@ -0,0 +1,700 @@
+        Simple method to add extra-parameters to a string
+    _lagrange_to_str(self, objective_list: ObjectiveList)
+        Convert Lagrange objective into an easy-to-read string
+    _mayer_to_str(self, objective_list: ObjectiveList)
+        Convert Mayer objective into an easy-to-read string  
+    _scaling_parameter(self, parameter: Parameter)
+        Take scaling into account for display task
+    _get_parameter_function_name(self, parameter: Parameter)
+        Get parameter function name (whether or not it is a custom function)
+    _analyze_nodes(self, phase_idx: int, constraint: Constraint)
+        Determine node index
+    """
+
+    def __init__(
+        self,
+        ocp,
+    ):
+        """
+        Parameters
+        ----------
+        ocp: OptimalControlProgram
+            A reference to the ocp
+        """
+
+        self.ocp = ocp
+
+    def _add_dict_to_str(self, _dict: dict):
+        """
+        Convert information contained in a dict to string
+
+        Parameters
+        ----------
+        _dict: dict
+            The dict to be converted
+        """
+
+        str_to_add = ""
+        for d in _dict:
+            str_to_add += f"{d}: {_dict[d]}{self._return_line}"
+        return str_to_add
+
+    def _add_extra_parameters_to_str(self, list_params: list, string: str):
+        """
+        Simple method to add extra-parameters to a string
+
+        Parameters
+        ----------
+        list_params: list
+            The list of parameters to add to the string
+        string: str
+            The string to be completed
+        """
+
+        for param in list_params:
+            string += f"{param}: {list_params[param]}{self._return_line}"
+        string += f"{self._return_line}"
+        return string
+
+    def _lagrange_to_str(self, objective_list: ObjectiveList):
+        """
+        Convert Lagrange objective into an easy-to-read string
+
+        Parameters
+        ----------
+        objective_list: ObjectiveList
+            The list of Lagrange objectives
+        """
+
+        objective_names = []
+        lagrange_str = ""
+        for objective in objective_list:
+            if len(objective) > 0:
+                obj = objective[0]["objective"]
+                if isinstance(obj.type, ObjectiveFcn.Lagrange):
+                    if obj.sliced_target is not None:
+                        if obj.quadratic:
+                            lagrange_str += f"({obj.name} - {obj.sliced_target}){self._squared}{self._return_line}"
+                        else:
+                            lagrange_str += f"{obj.name} - {obj.sliced_target}{self._return_line}"
+                    else:
+                        if obj.quadratic:
+                            lagrange_str += f"({obj.name}){self._squared}{self._return_line}"
+                        else:
+                            lagrange_str += f"{obj.name}{self._return_line}"
```

Comment:
Could you also add the test cases for these lines? I feel it is just adding 4 objective functions to your tests (actually two, one with quadratic=True and the same with quadratic=False)

Reply:
Done